### PR TITLE
`DataDictionary.getVersion()` - return version including service pack 

### LINKED ISSUE #381: Make `DataDictionary.getVersion()` return version including service pack
Follow-up to #258 where `getVersion()` was not changed for a minor QFJ release. But with 3.0.0 we should change this.

### PR DESCRIPTION
**Pull request by vinipx**
fixes #381 

@chrjohn please, review if it makes sense.

Also, should we keep the `setFullVersion`, `getFullVersion`, `setExtensionPackget`, `getExtensionPack` etc?

### CODE REVIEW

File: quickfixj-core/src/main/java/quickfix/DataDictionary.java [hunk 1]
```diff
@@ -882,15 +895,19 @@ private void checkValue(StringField field) throws IncorrectTagValue {
...
     private void checkHasValue(StringField field) {
         if (checkFieldsHaveValues && field.getValue().length() == 0) {
             throw new FieldException(SessionRejectReason.TAG_SPECIFIED_WITHOUT_A_VALUE,
                     field.getField());
         }
     }
 
-    /** Check if group count matches number of groups in **/
+    /**
+     * Check if group count matches number of groups in
```
Comment (chrjohn) [inline quickfixj-core/src/main/java/quickfix/DataDictionary.java:909]:
```suggestion
     * Check if group count matches number of groups in message.
```

**Comment by vinipx** [comment]
@chrjohn Please, review. I've introduced `servicePackVersion` to support implementation with minimal code changes. 
Let me know if its appropriate or we should think about better solution.

File: quickfixj-core/src/main/java/quickfix/DataDictionary.java [hunk 2]
```diff
@@ -177,10 +178,15 @@ private void setVersion(String beginString) {
     /**
      * Get the FIX major/minor version associated with this dictionary.
```
Comment (vinipx) [inline quickfixj-core/src/main/java/quickfix/DataDictionary.java:179]: potentially here we might add description about servicePack version too

**Comment by vinipx** [comment]
@chrjohn pls, review. I've added the servicepack reference to the method doc.

**Comment by chrjohn** [comment]
Hi @vinipx currently out of the office. I still need to check in more detail what `getVersion()` should return.

**Comment by vinipx** [comment]
> Hi @vinipx currently out of the office. I still need to check in more detail what `getVersion()` should return.

Sure, no rush. Thanks! We can elaborate it better, count on me.

**Comment by vinipx** [comment]
> Hi @vinipx ,
> 
> I thought about this and now am of the opinion that `getVersion()` should not return the full version.
> 
> Reasons for this:
> 
> 1. The `version` in the dictionary represents the `BeginString` from the FIX message. The `BeginString` does not contain service or extension pack.
> 2. The other QuickFIX variants (C++, .NET) also do not return service or extension pack there. Although we cannot be 100% compatible between the variants, I believe we should not change this.
> 
> Sorry, that you did some of the work already and sorry that I did not think this completely through beforehand. :-/
> 
> The original issue #258 also lists option 2 and 3 which are valid so I think we should leave `getVersion()` as it is now.

No problem at all - to be honest, I also found it a bit strange, but decided to proceed as a proposal only. 
Agreed - we can close this PR since there is already servicePack and fullversion available.

Thanks!

**Comment by chrjohn** [comment]
@vinipx 
Thank you for your understanding and contributions! :+1:

**Pull request closed without merging**